### PR TITLE
React router

### PR DESCRIPTION
**Pull request by AlaaShurrab**
creating the context file and updating it 
updating the other pages to use the is Auth context 
creating the logged out routes and the protected routes files and updating the router

### CODE REVIEW

File: client/src/pages/logIn/index.js [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
 import './style.css';
 
-const { Title } = Typography;
-
 const LoginPage = () => (
   <div className="login">
     <div className="header">
       <div className="loginBody">
-        <Title level={2}>log In</Title>
+        <CustomTitle text="log In" />
```
Comment (Mu7ammadAbed) [inline client/src/pages/logIn/index.js:11]:
```suggestion
        <CustomTitle text="Log In" />
```

File: client/src/index.js [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
 
 ReactDOM.render(
   <React.StrictMode>
     <Router>
-      <App />
+      <AuthProvider>
+        <App />
+      </AuthProvider>
     </Router>
   </React.StrictMode>,
```
Comment (Mu7ammadAbed) [inline client/src/index.js:16]: Remove the strict mode

File: client/src/index.css [hunk 3]
```diff
@@ -16,3 +16,10 @@ code {
     monospace;
 }
 
+#root{
+  min-height: 100vh;
+}
+
+.App{
+  min-height: 100vh;
+}
```
Comment (Mu7ammadAbed) [inline client/src/index.css:25]: new line here

File: client/src/Context/isAuthContext.js [hunk 4]
```diff
@@ -1,5 +1,35 @@
-import { createContext } from 'react';
+import React, { useContext, useState, createContext, useEffect } from 'react';
+import { element } from 'prop-types';
+import { get } from 'axios';
 
 const IsAuthContext = createContext(false);
 
+export const AuthProvider = ({ children }) => {
+  const authState = useState();
+  const [, setIsAuth] = authState;
```
Comment (Mu7ammadAbed) [inline client/src/Context/isAuthContext.js:9]:
```suggestion
  const [isAuth, setIsAuth] = useState();
```

File: client/src/Context/isAuthContext.js [hunk 5]
```diff
@@ -1,5 +1,35 @@
...
+        setIsAuth(true);
+      } catch ({ response }) {
+        setIsAuth(false);
+      }
+    };
+    checkAuth();
+  }, [setIsAuth]);
+
+  return (
+    <IsAuthContext.Provider value={authState}>
```
Comment (Mu7ammadAbed) [inline client/src/Context/isAuthContext.js:24]:
```suggestion
  }, [isAuth]);

  return (
    <IsAuthContext.Provider value={[isAuth, setIsAuth]}>
```

**Review by hassanelnajjar — APPROVED** [review]

File: client/src/pages/logIn/index.js (hunk 1 above)
Comment (AlaaShurrab) [reply client/src/pages/logIn/index.js:11]: done

File: client/src/index.js (hunk 2 above)
Comment (AlaaShurrab) [reply client/src/index.js:16]: done

File: client/src/index.css (hunk 3 above)
Comment (AlaaShurrab) [reply client/src/index.css:25]: done'

File: client/src/Context/isAuthContext.js (hunk 4 above)
Comment (AlaaShurrab) [reply client/src/Context/isAuthContext.js:9]: i don't need the isAuth in this file so I just get the setIsAuth

File: client/src/Context/isAuthContext.js (hunk 5 above)
Comment (AlaaShurrab) [reply client/src/Context/isAuthContext.js:24]: i think both ways work fine

**Review by hassanelnajjar — APPROVED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by Alaalser — APPROVED** [review]

**Pull request merged**